### PR TITLE
Add correlation id in response headers

### PR DESCRIPTION
**Pull request by agatty-chacrasoftware**

### CODE REVIEW

File: src/graphql/middleware/responseTransactionIdMiddleware.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+	next
+) => {
+	if (context.responseTranscationId) {
+		return next();
+	}
+
+	context.responseTranscationId = true;
+	try {
+		await next();
+		const txId = uuid4();
```
Comment (iskandar-chacrasoftware) [inline src/graphql/middleware/responseTransactionIdMiddleware.ts]: rename variable txId to transactionId

File: src/graphql/middleware/correlationIdMiddleware.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+export const CorrelationIdMiddleware: MiddlewareFn<any> = async (
+	{ context },
+	next
+) => {
+	if (context.responseCorrelationId) {
+		return next();
+	}
+
+	context.responseCorrelationId = true;
+	const correlationId = uuid4();
```
Comment (iskandar-chacrasoftware) [inline src/graphql/middleware/correlationIdMiddleware.ts:14]: needs spacing here

File: src/graphql/middleware/responseTransactionIdMiddleware.ts (hunk 1 above)
Comment (agatty-chacrasoftware) [reply src/graphql/middleware/responseTransactionIdMiddleware.ts]: This is being renamed as correlationId in the recent commits.

File: src/graphql/middleware/correlationIdMiddleware.ts (hunk 2 above)
Comment (agatty-chacrasoftware) [reply src/graphql/middleware/correlationIdMiddleware.ts:14]: Changes done.

**Pull request merged**